paasta_tools/api/views/instance.py

```diff
@@ -815,6 +815,36 @@ def instance_delay(request):
         return response
 
 
+@view_config(
+    route_name="service.instance.bounce_status", request_method="GET", renderer="json",
+)
+def bounce_status(request):
+    service = request.swagger_data.get("service")
+    instance = request.swagger_data.get("instance")
+    try:
+        instance_type = validate_service_instance(
+            service, instance, settings.cluster, settings.soa_dir
+        )
+    except NoConfigurationForServiceError:
+        error_message = (
+            "Deployment key %s not found.  Try to execute the corresponding pipeline if it's a fresh instance"
+            % ".".join([settings.cluster, instance])
+        )
+        raise ApiFailure(error_message, 404)
+    except Exception:
+        error_message = traceback.format_exc()
+        raise ApiFailure(error_message, 500)
+
+    if instance_type != "kubernetes":
```

Comment:
I think this should be `not pik.can_handle(instance_type)` instead. This will eval to `False` for all instance types running on k8s, like flink, cassandracluster, etc. not just kubernetes, because a lot of operator instances do rely on m-f-d. It's just that when the bounce, the operator rolls the pods, not s-k-j.

Reply:
I don't think they use `mark-for-deployment --watch` -- the current implementation only cares about kubernetes and marathon: https://github.com/Yelp/paasta/blob/master/paasta_tools/cli/cmds/mark_for_deployment.py#L1198